src/twisted/application/app.py

```diff
@@ -681,3 +690,17 @@ def startApplication(application, save):
         reactor.addSystemEventTrigger('after', 'shutdown', p.save, 'shutdown')
     reactor.addSystemEventTrigger('before', 'shutdown',
                                   service.IService(application).stopService)
+
+
+
+def exitWithSignal(sig):
+    """
+    Force the application to terminate with the specified signal by replacing
+    the signal handler with the default and sending the signal to ourselves.
+
+    @param sig:  Signal to use to terminate the process with C{os.kill}.
+    @type sig:  C{int}
+
+    """
+    signal.signal(sig, signal.SIG_DFL)
```

Comment:
> My concern is that, while the GNU C library certainly obeys the above behavior, I haven't found any spec/guarantee to indicate this would be consistent with other implementations. 

Correct.  There is no POSIX API for _constructing_ an exit code - only deconstructing it.  You probe the system to create some  values and then inspect them to see how the system composes the two pieces of information and then compose them yourself that way ... but ...  you could also just re-deliver the signal. :)